src/xcode/ENA/ENA/Source/Scenes/FileScanner/FileScannerCoordinator.swift

```diff
@@ -4,64 +4,136 @@
 
 import UIKit
 import PhotosUI
+import PDFKit
 
-class FileScannerCoordinator {
+class FileScannerCoordinator: NSObject, UIImagePickerControllerDelegate, UINavigationControllerDelegate, PHPickerViewControllerDelegate, UIDocumentPickerDelegate {
 	
 	// MARK: - Init
 
 	init(
 		_ parentViewController: UIViewController,
-		fileScannerViewModel: FileScannerProcessing,
+		viewModel: FileScannerProcessing,
 		qrCodeFound: @escaping (QRCodeResult) -> Void,
 		noQRCodeFound: @escaping () -> Void
 	) {
 		self.parentViewController = parentViewController
-		self.fileScannerViewModel = fileScannerViewModel
+		self.viewModel = viewModel
 		self.qrCodeFound = qrCodeFound
 		self.noQRCodeFound = noQRCodeFound
 	}
 
+	// MARK: - UIImagePickerControllerDelegate
+
+	func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [UIImagePickerController.InfoKey: Any]) {
+		finishedPickingImage()
+
+		DispatchQueue.global(qos: .background).async { [weak self] in
```

Comment:
```suggestion
		DispatchQueue.global(qos: . userInitiated).async { [weak self] in
```
Seems better fitting, don't you think?
https://developer.apple.com/documentation/dispatch/dispatchqos

Reply:
This is new, thanks for sharing @marcussc